### PR TITLE
br: support reset_tiflash after ebs restoration

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40208

Problem Summary: resettle tiflash replica after ebs restoration. This PR associate with https://github.com/pingcap/tidb-operator/pull/4812.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fengou1 — APPROVED** [review]

**Comment by fengou1** [comment]
please help to fix ut broken. @3pointer

**Review by WangLe1321 — APPROVED** [review]

File: br/pkg/restore/tiflashrec/tiflash_recorder.go [hunk 1]
```diff
@@ -79,6 +79,43 @@ func (r *TiFlashRecorder) Rewrite(oldID int64, newID int64) {
...
+		if err != nil {
+			log.Warn("Failed to generate the alter table spec", logutil.ShortError(err), zap.Any("replica", replica))
+			return
+		}
+		items = append(items, fmt.Sprintf(
+			"ALTER TABLE %s %s",
+			utils.EncloseDBAndTable(schema.Name.O, table.Meta().Name.O),
+			altTableSpec),
+		)
+		altTableSpec, err = alterTableSpecOf(replica, false)
```
Comment (joccau) [inline br/pkg/restore/tiflashrec/tiflash_recorder.go:105]: Add a comment about the reason for setting the tiflash-count=0 and then set normal tiflash-count.

**Review by joccau — APPROVED** [review]
LGTM rest.

**Pull request merged**